### PR TITLE
SAGA processing: update required version and FreeBSD patches

### PR DESCRIPTION
**Pull request by lbartoletti**
## Description
The PR contains two commits:
- the first one fixing FreeBSD support
- the second is the SAGA version, which for FreeBSD users seems to work well with newer versions (7.1.1 on my system). cc @rhurlin for testing.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nyalldawson** [comment]
> * the second is the SAGA version, which for FreeBSD users seems to work well with newer versions (7.1.1 on my system). cc @rhurlin for testing.

We can't do this -- the API changes version by version with SAGA, so we ONLY target the SAGA LTR release (which is stuck at the mega-outdated 2.3 release). You'll need to convince the SAGA upstream developers to publish a new LTR release before we can pursue this change.

**Comment by roya0045** [comment]
> > * the second is the SAGA version, which for FreeBSD users seems to work well with newer versions (7.1.1 on my system). cc @rhurlin for testing.
> 
> We can't do this -- the API changes version by version with SAGA, so we ONLY target the SAGA LTR release (which is stuck at the mega-outdated 2.3 release). You'll need to convince the SAGA upstream developers to publish a new LTR release before we can pursue this change.

If the SAGA team does not seem to be inclined to publish a LTR and since we are on version 7.2. It might be possible that the use of LTR might not be a thing to them. Would it be possible to adapt qgis to support normal releases of SAGA? Or is this a development directive to only use LTR releases?

**Comment by nyalldawson** [comment]
>  Would it be possible to adapt qgis to support normal releases of SAGA? Or is this a development directive to only use LTR releases?

This was the older approach, and it was a maintenance nightmare. The SAGA API changes regularly even in minor versions, and the only way to upgrade the Processing SAGA descriptions was for someone to manually detect the differences and upgrade the descriptions. Even now are still encountering description files which are out of date with the 2.3 release. There's just not the people power available to maintain anything more than the LTR release.

There's also the issue that distros like debian/ubuntu/Fedora only publish the LTR releases (and the current LTR doesn't work reliably here without downstream patches -- e.g. the Fedora package is completely broken due to https://bugzilla.redhat.com/show_bug.cgi?id=1494545)

Really, the ideal fix here is for SAGA to publish a new LTR and keep it maintained.

File: python/plugins/processing/algs/saga/SagaAlgorithmProvider.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def setActive(self, active):
 
     def canBeActivated(self):
         version = SagaUtils.getInstalledVersion(True)
-        if version is not None and version.startswith(REQUIRED_VERSION):
+        if version is not None and version > REQUIRED_VERSION:
```
Comment (m-kuhn) [inline python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]:
```suggestion
        if version is not None and version >= REQUIRED_VERSION:
```

**Comment by gioman** [comment]
> This was the older approach, and it was a maintenance nightmare.

I can confirm that... as one of the guys that tried to keep updated the SAGA description files for multiple SAGA versions...

**Comment by lbartoletti** [comment]
@nyalldawson @gioman I understand your point of view. Thank you for your work on SAGA.
I will close this PR to create another PR specific to FreeBSD upstream patches.

@rhurlin Rainer, do you have the possibility to request LTR versions from SAGA developers? :)

File: python/plugins/processing/algs/saga/SagaAlgorithmProvider.py (hunk 1 above)
Comment (rhurlin) [reply python/plugins/processing/algs/saga/SagaAlgorithmProvider.py]: Sorry for the late response.
On FreeBSD, the latter patch makes newer versions, like the recent 7.2.0 usable with the FreeBSD port of QGIS, see under https://www.freshports.org/graphics/qgis. 

Of course, some of the newer SAGA GIS modules are not documented within QGIS processing until now, but on the other hand, many many bug fixes of newer SAGA GIS versions are included.

**Comment by rhurlin** [comment]
@lbartoletti Loïc, I am not a member of the SAGA GIS development team, but I will try :)

**Comment by lbartoletti** [comment]
@nyalldawson I revert saga version on this PR as requested. Thanks

**Comment by nyalldawson** [comment]
Is this redundant now?

**Pull request closed without merging**